googletest/include/gtest/internal/gtest-internal.h

```diff
@@ -307,7 +307,7 @@ class FloatingPoint {
   // around may change its bits, although the new value is guaranteed
   // to be also a NAN.  Therefore, don't expect this constructor to
   // preserve the bits in x when x is a NAN.
-  explicit FloatingPoint(const RawType& x) { u_.value_ = x; }
+  explicit FloatingPoint(const RawType& x) : u_() { u_.value_ = x; }
```

Comment:
> What does it mean to default-construct a union of RawType and Bits?
> I am not sure of the .. portability, or legality of 'u_()'.

There is a good explanation: http://stackoverflow.com/questions/321351/initializing-a-union-with-a-non-trivial-constructor

From C++03, 12.1 Constructors, pg 190

The implicitly-defined default constructor performs the set of initializations of the class that would be performed by a user-written default constructor for that class with an empty mem-initializer-list (12.6.2) and an empty function body.

From C++03, 8.5 Initializers, pg 145

To default-initialize an object of type T means:

if T is a non-POD class type (clause 9), the default constructor for T is called (and the initialization is ill-formed if T has no accessible default constructor);
if T is an array type, each element is default-initialized;
otherwise, the object is zero-initialized.
To zero-initialize an object of type T means:

if T is a scalar type (3.9), the object is set to the value of 0 (zero) converted to T;
if T is a non-union class type, each non static data member and each base-class subobject is zero-initialized;
**if T is a union type, the object’s first named data member is zero-initialized;**
if T is an array type, each element is zero-initialized;
if T is a reference type, no initialization is performed.

> I am not sure of the usefulness .. of 'u_()'.

OK, I may add `explicit FloatingPointUnion(const RawType& v)` constructor to keep only one assignment